awx/main/tasks.py

```diff
@@ -769,43 +770,50 @@ def build_passwords(self, instance, **kwargs):
             '': '',
         }
 
-    def build_extra_vars_file(self, vars, **kwargs):
-        handle, path = tempfile.mkstemp(dir=kwargs.get('private_data_dir', None))
+    def build_extra_vars_file(self, instance, private_data_dir, passwords, display=False):
+        '''
+        Build ansible yaml file filled with extra vars to be passed via -e@file.yml
+        '''
+
+    def _write_extra_vars_file(self, private_data_dir, vars, safe_dict={}):
+        env_path = os.path.join(private_data_dir, 'env')
+        try:
+            os.mkdir(env_path, stat.S_IREAD | stat.S_IWRITE | stat.S_IEXEC)
+        except OSError as e:
+            if e.errno != errno.EEXIST:
+                raise
+
+        path = os.path.join(env_path, 'extravars')
+        handle = os.open(path, os.O_RDWR | os.O_CREAT, stat.S_IREAD | stat.S_IWRITE)
         f = os.fdopen(handle, 'w')
         if settings.ALLOW_JINJA_IN_EXTRA_VARS == 'always':
             f.write(yaml.safe_dump(vars))
         else:
-            f.write(safe_dump(vars, kwargs.get('safe_dict', {}) or None))
+            f.write(safe_dump(vars, safe_dict))
         f.close()
         os.chmod(path, stat.S_IRUSR)
         return path
 
-    def add_ansible_venv(self, venv_path, env, add_awx_lib=True, **kwargs):
+    def add_ansible_venv(self, venv_path, env, isolated=False):
         env['VIRTUAL_ENV'] = venv_path
         env['PATH'] = os.path.join(venv_path, "bin") + ":" + env['PATH']
         venv_libdir = os.path.join(venv_path, "lib")
 
-        if not kwargs.get('isolated', False) and not os.path.exists(venv_libdir):
+        if isolated and not os.path.exists(venv_libdir):
             raise RuntimeError(
                 'a valid Python virtualenv does not exist at {}'.format(venv_path)
             )
         env.pop('PYTHONPATH', None)  # default to none if no python_ver matches
-        for version in os.listdir(venv_libdir):
-            if fnmatch.fnmatch(version, 'python[23].*'):
-                if os.path.isdir(os.path.join(venv_libdir, version)):
-                    env['PYTHONPATH'] = os.path.join(venv_libdir, version, "site-packages") + ":"
-                    break
-        # Add awx/lib to PYTHONPATH.
-        if add_awx_lib:
-            env['PYTHONPATH'] = env.get('PYTHONPATH', '') + self.get_path_to('..', 'lib') + ':'
+        if os.path.isdir(os.path.join(venv_libdir, "python2.7")):
+            env['PYTHONPATH'] = os.path.join(venv_libdir, "python2.7", "site-packages") + ":"
         return env
 
     def add_awx_venv(self, env):
         env['VIRTUAL_ENV'] = settings.AWX_VENV_PATH
         env['PATH'] = os.path.join(settings.AWX_VENV_PATH, "bin") + ":" + env['PATH']
         return env
 
-    def build_env(self, instance, **kwargs):
+    def build_env(self, instance, private_data_dir, isolated, private_data_files={}):
```

Comment:
mutable default arguments are frowned upon, but shouldn't matter here since it's only read, never changed.

Reply:
It's only read _at this moment_.  Let's default this to `None` and do the check in the method.